pkg/serverless/invocationlifecycle/trace.go

```diff
@@ -23,17 +27,36 @@ type executionStartInfo struct {
 	startTime time.Time
 	traceID   uint64
 	spanID    uint64
+	parentID  uint64
+}
+type invocationPayload struct {
+	Headers map[string]string `json:"headers"`
 }
 
 // currentExecutionInfo represents information from the start of the current execution span
 var currentExecutionInfo executionStartInfo
 
 // startExecutionSpan records information from the start of the invocation.
 // It should be called at the start of the invocation.
-func startExecutionSpan(startTime time.Time) {
+func startExecutionSpan(startTime time.Time, rawPayload string) {
```

Comment:
I would update this function to accept an already-parsed `invocationPayload` as a parameter rather than a rawPayload `string`. Then you can move the conversion of the payload outside of this function and keep it totally separate. Having this function do less and separating parts of it out into other functions will make it easier to test over time.

Reply:
This became too complicated for the current task. Going to leave this for another PR.